### PR TITLE
fix(files_sharing): Add one string for every share type

### LINKED ISSUE #42179: [Bug]: Missing space char in german translation file for "Share with " -> "Teilen mit" causing lingual issue
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

Problem caused by missing " " in translation files in /apps/files_sharing/l10n/de*.js and de*.json

2023-12-12_17h34_23

### Steps to reproduce

1. Go to share view of any shared folder
2. Click on the "three-dotts-button" right beside the persons name to see details of sharing

### Expected behavior

Instead of "Teilen mitPascal" it should be "Teilen mit Pascal"

### Installation method

Community Docker image

### Nextcloud Server version

28

### Operating system

None

### PHP engine version

None

### Web server

None

### Database engine version

None

### Is this bug present after an update or on a fresh install?

None

### Are you using the Nextcloud Server Encryption module?

None

### What user-backends are you using?

- [ ] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

_No response_

### List of activated Apps

_No response_

### Nextcloud Signing status

_No response_

### Nextcloud Logs

_No response_

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by susnux**
* Resolves: https://github.com/nextcloud/server/issues/42179

## Summary
Do not assume any order of words for translatable string  but be explicit and add a string for every type.

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Review by rakekniven — APPROVED** [review]
I cannot judge the code 100% but the idea is covered. Great.

File: apps/files_sharing/src/views/SharingDetailsTab.vue [hunk 1]
```diff
@@ -305,24 +305,24 @@ export default {
...
+			case this.SHARE_TYPES.SHARE_TYPE_REMOTE:
+				return t('files_sharing', 'Share with remote')
+			case this.SHARE_TYPES.SHARE_TYPE_REMOTE_GROUP:
+				return t('files_sharing', 'Share with remote group')
+			case this.SHARE_TYPES.SHARE_TYPE_GUEST:
+				return t('files_sharing', 'Share with guest')
 			}
 
-			return title
+			return t('files_sharing', 'Share with')
```
Comment (emoral435) [inline apps/files_sharing/src/views/SharingDetailsTab.vue:325]: More of a nitpick than anything, but can't we have the `default` section of a switch case have the last return statement?

> return t('files_sharing', 'Share with')

This part ^

**Review by emoral435 — APPROVED** [review]
Looks good to me : ) love seeing switch statements 👌

File: apps/files_sharing/src/views/SharingDetailsTab.vue (hunk 1 above)
Comment (susnux) [reply apps/files_sharing/src/views/SharingDetailsTab.vue]: I like this :+1:

**Pull request merged**